src/platform/packages/private/kbn-esql-editor/src/esql_editor.tsx

```diff
@@ -1126,21 +1057,31 @@ const ESQLEditorInternal = function ESQLEditor({
                       await addLookupIndicesDecorator();
                     }
 
-                    // When both inline and suggestion widget are visible,
-                    // we want Tab to accept inline suggestions,
-                    // so we need to unbind the default suggestion widget behavior
-                    monaco.editor.addKeybindingRule({
-                      keybinding: monaco.KeyCode.Tab,
-                      command: '-acceptSelectedSuggestion',
-                      when: 'suggestWidgetHasFocusedSuggestion && suggestWidgetVisible && textInputFocus && inlineSuggestionVisible',
+                    // Register custom commands
+                    const commandDisposables = registerCustomCommands({
+                      application,
+                      uiActions,
+                      telemetryService,
+                      editor1: editor1 as React.RefObject<monaco.editor.IStandaloneCodeEditor>,
+                      fixedQuery,
+                      esqlVariables,
+                      controlsContext,
+                      openTimePickerPopover,
                     });
 
-                    // Add explicit binding for Tab to accept inline suggestions when they're visible
-                    monaco.editor.addKeybindingRule({
-                      keybinding: monaco.KeyCode.Tab,
-                      command: 'editor.action.inlineSuggest.commit',
-                      when: 'inlineSuggestionVisible && textInputFocus',
-                    });
+                    // Add editor key bindings
+                    addEditorKeyBindings(editor, onQuerySubmit, setIsVisorOpen, isVisorOpen);
+
+                    // Store disposables for cleanup
+                    const currentEditor = editor1.current;
+                    if (currentEditor) {
+                      if (!editorCommandDisposables.current.has(currentEditor)) {
+                        editorCommandDisposables.current.set(currentEditor, commandDisposables);
```

Comment:
Do you want to dispose of kwy binding commands as well? `addCommand `does not have disposables. `addAction `does.

Reply:
The addCommand is getting disposed when you dispose the editor. The problem here was that it was being created on every keypress. The same with the registerCommand which has the additional problem that is not disposed automatically. This is why I am disposing it manually.